backend/oasst_backend/prompt_repository.py

```diff
@@ -308,18 +308,22 @@ def insert_post(
         self.db.refresh(post)
         return post
 
-    def insert_reaction(self, post_id: UUID, payload: db_payload.ReactionPayload) -> PostReaction:
+    def upsert_reaction(self, post_id: UUID, payload: db_payload.ReactionPayload) -> PostReaction:
         if self.person_id is None:
             raise ValueError("User required")
 
         container = PayloadContainer(payload=payload)
-        reaction = PostReaction(
+        new_reaction = PostReaction(
```

Comment:
It would make sense to move the construction of the new `PostReaction` obj into the `if existing_reaction is not None:` block.

Reply:
Done :+1: